### PR TITLE
Prepare for v3.0.4 release

### PR DESCRIPTION
**Pull request by jsquyres**
Refs #5987 (v3.0.x checklist).

### CODE REVIEW

**Comment by jsquyres** [comment]
@bwbarrett After making the v3.1.x version of this PR, I came back to this PR and updated the R number for all the `.so` files.  See the commit message for details.

**Comment by jsquyres** [comment]
@bwbarrett The NEWS in this PR assumes we get a fix for #6436.

File: README [hunk 1]
```diff
@@ -64,7 +64,7 @@ Much, much more information is also available in the Open MPI FAQ:
 ===========================================================================
 
 The following abbreviated list of release notes applies to this code
-base as of this writing (March 2017):
+base as of this writing (February 2019):
```
Comment (bwbarrett) [inline README]: sad trombone :)

Comment (jsquyres) [reply README]: Can fix momentarily.  Will just have to wait for CI.

**Pull request merged**